feat: upgrade to quince

ziafazal opened the pull request:
Changes related to quince upgrade.

Commit:
docs: change quickstart to launch

This is due to renaming quickstart to launch in tutor.

Commit:
Merge remote-tracking branch 'origin/master' into nightly

Commit:
Merge remote-tracking branch 'origin/master' into nightly

Commit:
Merge remote-tracking branch 'origin/master' into nightly

Commit:
Merge remote-tracking branch 'origin/master' into nightly

Commit:
Merge remote-tracking branch 'origin/master' into nightly

Commit:
Merge remote-tracking branch 'origin/master' into nightly

Commit:
Merge remote-tracking branch 'origin/master' into nightly

Commit:
Merge remote-tracking branch 'origin/master' into nightly

Commit:
Merge remote-tracking branch 'origin/master' into nightly

Commit:
Merge remote-tracking branch 'origin/master' into nightly

Commit:
Merge remote-tracking branch 'origin/master' into nightly

Commit:
Merge remote-tracking branch 'origin/master' into nightly

Commit:
Merge remote-tracking branch 'origin/master' into nightly

Commit:
Merge remote-tracking branch 'origin/master' into nightly

regisb committed:
feat: label nightly version

This is to address https://github.com/overhangio/tutor-mfe/issues/122
As a consequence of this change, images will be tagged with a "-nightly"
suffix. Next, we'll probably have to build them periodically in CI.

regisb committed:
fix: nightly package version

The package version number may not include the "-nightly" suffix.
Otherwise, installation fails with:

   setuptools.extern.packaging.version.InvalidVersion: Invalid version: '15.0.7-nightly'

regisb committed:
Merge branch 'master' into nightly

Commit:
Merge remote-tracking branch 'origin/master' into nightly

Commit:
Merge remote-tracking branch 'origin/master' into nightly

Commit:
Merge remote-tracking branch 'origin/master' into nightly

ziafazal committed:
feat: upgrade to quince

Commit:
Merge remote-tracking branch 'origin/master' into nightly

Commit:
Merge remote-tracking branch 'origin/master' into nightly

regisb committed:
Merge branch 'master' into nightly

regisb commented on tutorecommerce/templates/ecommerce/build/ecommerce/Dockerfile:
```diff
@@ -48,7 +48,7 @@ RUN {% if is_buildkit_enabled() %}--mount=type=cache,target=/openedx/.cache/pip,
+    setuptools==68.2.2 pip==23.2.1 wheel==0.41.2
 
 # Install a recent version of nodejs
 # https://pypi.org/project/nodeenv
```
Please take this opportunity to upgrade to nodeenv 1.8.0.

Commit:
Merge remote-tracking branch 'origin/master' into nightly

Commit:
Merge remote-tracking branch 'origin/master' into nightly

regisb commented:
I'm getting the following error during build:

```
 > [payment-common 3/5] RUN --mount=type=cache,target=/root/.npm,sharing=shared npm clean-install --no-audit --no-fund --registry=https://registry.npmjs.org/:
7.330 npm notice 
7.338 npm notice New major version of npm available! 9.5.1 -> 10.2.4
7.338 npm notice Changelog: <https://github.com/npm/cli/releases/tag/v10.2.4>
7.338 npm notice Run `npm install -g npm@10.2.4` to update!
7.338 npm notice 
7.376 npm ERR! code EUSAGE
7.379 npm ERR! 
7.379 npm ERR! `npm ci` can only install packages when your package.json and package-lock.json or npm-shrinkwrap.json are in sync. Please update your lock file with `npm install` before continuing.
7.379 npm ERR! 
7.379 npm ERR! Invalid: lock file's @babel/types@7.22.5 does not satisfy @babel/types@7.23.5
7.379 npm ERR! Invalid: lock file's @babel/template@7.22.5 does not satisfy @babel/template@7.22.15
7.379 npm ERR! Invalid: lock file's @babel/code-frame@7.22.5 does not satisfy @babel/code-frame@7.23.5
7.379 npm ERR! Invalid: lock file's @babel/parser@7.22.5 does not satisfy @babel/parser@7.23.5
7.379 npm ERR! Invalid: lock file's @babel/highlight@7.22.5 does not satisfy @babel/highlight@7.23.4
7.379 npm ERR! Invalid: lock file's @babel/helper-validator-identifier@7.22.5 does not satisfy @babel/helper-validator-identifier@7.22.20
7.379 npm ERR! Invalid: lock file's @babel/helper-environment-visitor@7.22.5 does not satisfy @babel/helper-environment-visitor@7.22.20
7.379 npm ERR! Invalid: lock file's @babel/helper-split-export-declaration@7.22.5 does not satisfy @babel/helper-split-export-declaration@7.22.6
7.380 npm ERR! Invalid: lock file's @babel/helper-string-parser@7.22.5 does not satisfy @babel/helper-string-parser@7.23.4
7.380 npm ERR! 
7.380 npm ERR! Clean install a project
7.380 npm ERR! 
7.380 npm ERR! Usage:
7.380 npm ERR! npm ci
7.380 npm ERR! 
7.380 npm ERR! Options:
7.380 npm ERR! [-S|--save|--no-save|--save-prod|--save-dev|--save-optional|--save-peer|--save-bundle]
7.380 npm ERR! [-E|--save-exact] [-g|--global]
7.380 npm ERR! [--install-strategy <hoisted|nested|shallow|linked>] [--legacy-bundling]
7.380 npm ERR! [--global-style] [--omit <dev|optional|peer> [--omit <dev|optional|peer> ...]]
7.380 npm ERR! [--strict-peer-deps] [--no-package-lock] [--foreground-scripts]
7.380 npm ERR! [--ignore-scripts] [--no-audit] [--no-bin-links] [--no-fund] [--dry-run]
7.380 npm ERR! [-w|--workspace <workspace-name> [-w|--workspace <workspace-name> ...]]
7.380 npm ERR! [-ws|--workspaces] [--include-workspace-root] [--install-links]
7.380 npm ERR! 
7.380 npm ERR! aliases: clean-install, ic, install-clean, isntall-clean
7.380 npm ERR! 
7.380 npm ERR! Run "npm help ci" for more info
7.382 
7.382 npm ERR! A complete log of this run can be found in:
7.382 npm ERR!     /root/.npm/_logs/2023-11-29T16_00_59_465Z-debug-0.log
------
Dockerfile:577
--------------------
 575 |     ENV CPPFLAGS=-DPNG_ARM_NEON_OPT=0
 576 |     ENV PACT_SKIP_BINARY_INSTALL=true
 577 | >>> RUN --mount=type=cache,target=/root/.npm,sharing=shared npm clean-install --no-audit --no-fund --registry=$NPM_REGISTRY
 578 |     
 579 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c npm clean-install --no-audit --no-fund --registry=$NPM_REGISTRY" did not complete successfully: exit code: 1
Error: Command failed with status 1: docker buildx build --tag=docker.io/overhangio/openedx-mfe:17.0.0 --no-cache --output=type=docker /home/tutor/.local/share/tutor/env/plugins/mfe/build/mfe

```

Commit:
Merge remote-tracking branch 'origin/master' into nightly

Commit:
Merge remote-tracking branch 'origin/master' into nightly

regisb commented on tutorecommerce/plugin.py:
```diff
@@ -60,7 +60,7 @@
         "MFE_APP": {
             "name": "orders",
             "repository": "https://github.com/edx/frontend-app-ecommerce",
-            "port": 1996,
```
I address this issue in this PR: https://github.com/overhangio/tutor-ecommerce/pull/57/files
We should merge it (I'll do it now), then rebase the quince branch on top of nightly.

Commit:
Merge remote-tracking branch 'origin/master' into nightly

Commit:
Merge remote-tracking branch 'origin/master' into nightly

regisb committed:
local.overhang.io -> local.edly.io

This is related to https://github.com/overhangio/tutor/issues/945

Commit:
Merge remote-tracking branch 'origin/master' into nightly

regisb committed:
Merge branch 'master' into nightly

regisb committed:
fix: MFE repo urls

Repo urls must end with ".git". Otherwise `ADD --keep-git-dir` fails to
clone the repo.

Commit:
Merge remote-tracking branch 'origin/master' into nightly

Commit:
Merge remote-tracking branch 'origin/master' into nightly
